packages/editor/src/components/rich-text/format-toolbar/link-container.js

```diff
@@ -19,95 +23,205 @@ import { filterURLForDisplay } from '../../../utils/url';
+		super( ...arguments );
+
+		this.editLink = this.editLink.bind( this );
+		this.submitLink = this.submitLink.bind( this );
+		this.onKeyDown = this.onKeyDown.bind( this );
+		this.onChangeInputValue = this.onChangeInputValue.bind( this );
+		this.toggleLinkSettingsVisibility = this.toggleLinkSettingsVisibility.bind( this );
+		this.setLinkTarget = this.setLinkTarget.bind( this );
+
+		this.state = {
+			inputValue: link && link.attributes && link.attributes.href ? null : '',
+			opensInNewWindow: null,
+			lastSelection: selection,
+			selectionKey: 0,
+		};
+	}
+
+	static getDerivedStateFromProps( { link, selection }, state ) {
+		if ( selection === state.lastSelection ) {
+			return null;
+		}
+
+		return {
+			inputValue: link.attributes && link.attributes.href ? null : '',
+			opensInNewWindow: null,
+			lastSelection: selection,
+			selectionKey: state.selectionKey + 1,
+		};
+	}
+
+	onKeyDown( event ) {
+		if ( event.keyCode === ESCAPE ) {
+			event.stopPropagation();
+			this.props.removeFormat( 'a' );
+			this.setState( {
+				inputValue: null,
+				opensInNewWindow: null,
+			} );
+		}
+
+		if ( [ LEFT, DOWN, RIGHT, UP, BACKSPACE, ENTER ].indexOf( event.keyCode ) > -1 ) {
+			// Stop the key event from propagating up to maybeStartTyping in BlockListBlock.
+			event.stopPropagation();
+		}
+	}
+
+	onChangeInputValue( inputValue ) {
+		this.setState( { inputValue } );
+	}
+
+	toggleLinkSettingsVisibility() {
+		this.setState( ( state ) => ( { settingsVisible: ! state.settingsVisible } ) );
+	}
+
+	setLinkTarget( opensInNewWindow ) {
+		// Apply now if URL is not being edited.
+		if ( this.state.opensInNewWindow === null ) {
+			const { link } = this.props;
+
+			if ( opensInNewWindow ) {
+				link.attributes.target = '_blank';
+				link.attributes.rel = 'noreferrer noopener';
+			} else {
+				delete link.attributes.target;
+				delete link.attributes.rel;
+			}
+
+			this.props.applyFormat( link );
+		} else {
+			this.setState( { opensInNewWindow } );
+		}
+	}
+
+	editLink( event ) {
+		const { link } = this.props;
+
+		this.setState( {
+			inputValue: link.attributes.href,
+			opensInNewWindow: link.attributes.target === '_blank',
+		} );
+
+		event.preventDefault();
+	}
+
+	submitLink( event ) {
+		const { link } = this.props;
+		const { inputValue, opensInNewWindow } = this.state;
+		const href = prependHTTP( inputValue );
+		const format = {
+			type: 'a',
+			attributes: {
+				href,
+			},
+		};
+
+		if ( opensInNewWindow ) {
+			format.attributes.target = '_blank';
+			format.attributes.rel = 'noreferrer noopener';
+		}
+
+		this.props.applyFormat( format );
+
+		this.setState( {
+			inputValue: null,
+			opensInNewWindow: null,
+		} );
+
+		if ( ! link ) {
+			this.props.speak( __( 'Link added.' ), 'assertive' );
+		}
+
+		event.preventDefault();
+	}
+
+	render() {
+		const { link } = this.props;
+		const { inputValue, settingsVisible, opensInNewWindow, selectionKey } = this.state;
+
+		if ( ! link ) {
```

Comment:
Isn't it handled also outside the component?

Reply:
Yes, can be removed. 👍